src/components/Tooltip/Tooltip.tsx

```diff
@@ -30,18 +33,46 @@ type Position =
   | "top-left"
   | "top-right";
 
-type Props = {
+export type Props = {
+  /**
+   * Whether the tooltip should adjust to fit in the screen.
+   */
+  autoAdjust?: boolean;
+  /**
+   * An optional class to apply to the wrapping element.
+   */
+  className?: string;
+  /**
+   * Element on which to apply the tooltip.
+   */
   children: ReactNode;
+  /**
+   * Whether the tooltip should follow the mouse.
+   */
+  followMouse?: boolean;
+  /**
+   * Message to display when the element is hovered.
+   */
   message?: string;
+  /**
+   * Position of the tooltip relative to the element.
+   */
   position?: Position;
+  /**
+   * An optional class to apply to the tooltip message element.
+   */
+  tooltipClassName?: string;
```

Comment:
Is this strictly necessary with the `className` prop above? As it should still be able to be targeted from that class?

I'm just seeing a lot of prop creep here so thinking of reasonable ways to shrink the api surface area.

Reply:
The actual tooltip gets rendered inside a portal which hoists it to the top level of the DOM (body) so it no longer exists inside the wrapping element. This class has been added so you can still target the message if needed.